psydac/api/feec.py

```diff
@@ -288,6 +300,39 @@ def derivatives(self, kind='femlinop'):
         elif kind == 'linop': 
             return tuple(b_diff.linop for b_diff in self._derivatives)
     
+    #--------------------------------------------------------------------------
+    def dirichlet_projectors(self, kind='femlinop'):
+        """
+        Returns operators that apply the correct Dirichlet boundary conditions.
+
+        Parameters
+        ----------
+        kind : str
+            The kind of the projector, can be 'femlinop' or 'linop'.
+            - 'femlinop' returns a psydac FemLinearOperator (default)
+            - 'linop' returns a psydac LinearOperator
+
+        Returns
+        -------
+        d_projectors : list
+            List of <psydac.fem.basic.FemLinearOperator> or <psydac.linalg.basic.LinearOperator>
+            The Dirichlet boundary projectors of each space and in desired form.
+
+        Notes
+        -----
+        See examples/vector_potential_3d.py for a use case of these operators in LinearOperator form.
+        
+        """
+        assert kind in ('femlinop', 'linop')
+
+        from psydac.fem.projectors import DirichletBoundaryProjector
+        d_projectors = [DirichletBoundaryProjector(Vh) for Vh in self.spaces[:-1]]
```

Comment:
Should we add these projectors to the derham class as an attribute? I would suggest to do it similarly to the conforming projectors. (The FemLinearoperator is stored as a tuple the first time this function is called)

Reply:
I agree with Frederik